### PR TITLE
Fixed: Mobile Navigation Menu

### LINKED ISSUE #2464: Navbar Close on link click
### Description

We can close the navbar ( mobile view ) When the routes are changed or A link is clicked

image

### Screenshots

_No response_

### Additional information

_No response_

### LINKED ISSUE #3191: Mobile (View) Navigation Menu changes required
### Description

- The hamburger menu should turn into a close icon when it is open(the SVG for the close icon is present in the code but is not implemented)
- When we click on the EddieHub Icon, it redirects us to the home page but doesn't close the mobile navigation menu.
- When we click outside the navigation menu, the mobile navigation menu should close.
- On clicking the hamburger menu it has a white outline around it, which doesn't look good.

### Issues

https://user-images.githubusercontent.com/116095538/211910355-5d1b408f-1857-4359-b6d3-fbc8c7282ff0.mp4

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by swagataroy30**
## Fixes Issue

Closes #3191 and Closes #2464 

## Changes proposed

- Added Close Icon when menu is open.
- When clicked on EddieHub Logo it closes the mobile navigation menu (if menu is opened)
- When clicked anywhere outside the mobile navigation menu, the navigation menu closes.
- Removed the outline from hamburger menu
- When the navigation links are clicked, it closes the mobile navigation menu.

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots

https://user-images.githubusercontent.com/116095538/211923197-b1699363-6e00-45f6-a8eb-f34faabb8797.mp4

## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/3202"></a>

### CODE REVIEW

File: components/navbar/Navbar.js [hunk 1]
```diff
@@ -124,7 +143,7 @@ export default function Navbar() {
           </div>
         </div>
 
-        <div className={`${!isOpen && "hidden"} md:hidden`} id="mobile-menu">
+        <div className={`${!isOpen && "hidden"} md:hidden`} id="mobile-menu" onClick={() => setIsOpen(false)}>
```
Comment (Aadarsh805) [inline components/navbar/Navbar.js:146]: currently clicking here will also close it, you can move the onClick function to the inner div 
image

**Review by Aadarsh805 — CHANGES_REQUESTED** [review]

**Comment by Aadarsh805** [comment]
Everything else works great 👍

File: components/navbar/Navbar.js (hunk 1 above)
Comment (swagataroy30) [reply components/navbar/Navbar.js:146]: I am sorry for that.It's my bad. Thanks for spotting it. I will be making the following changes.

**Comment by swagataroy30** [comment]
Hey @Aadarsh805 i did make the following changes ... could you check it once?

**Review by Aadarsh805 — DISMISSED** [review]
yep this would work

**Comment by ChinmayMhatre** [comment]
#2464 already has a PR fixing its issue.

**Comment by ChinmayMhatre** [comment]
Please have a discussion on the issue before creating a PR

**Comment by ChinmayMhatre** [comment]
> Screenshot from 2023-01-12 18-27-27
> 
> Actually I thought the PR on that issue was closed, my bad 😅. But this PR also resolves other issues mentioned in #3191 and that specific issue just kind of got into this fix. Sorry for not noticing.

Hey no worries, just the thing is if multiple PR's are created eddie and the maintainers have to solve the merge conflicts. Could you check which changes have been done in the other PR and keep the rest in your's? Thank you! 🙌🏻

**Comment by swagataroy30** [comment]
So should i be including the changes made in that PR to fix the issue #2464 in this PR? 
Screenshot from 2023-01-12 20-36-56
in the above snap, this PR includes this change to fix that issue, so should i remove this change and include the changes made in that PR?  Sorry is that what you said @ChinmayMhatre 😅? i didn't get it.

**Comment by eddiejaoude** [comment]
Great collaboration everyone 💪 

I think this merge PR https://github.com/EddieHubCommunity/LinkFree/pull/2507 was a bit different, it was for when a link is clicked it closes

@swagataroy30 could you update your branch from `main` and resolve conflicts. Thanks

**Comment by swagataroy30** [comment]
Done @eddiejaoude :+1: ...  is it all right?

**Review by eddiejaoude — APPROVED** [review]
Look good 👍

**Pull request merged**